libshaderc/include/shaderc.h

```diff
@@ -24,12 +24,26 @@ extern "C" {
 #include <stdint.h>
 
 typedef enum {
-  shaderc_glsl_vertex_shader,
-  shaderc_glsl_fragment_shader,
-  shaderc_glsl_compute_shader,
-  shaderc_glsl_geometry_shader,
-  shaderc_glsl_tess_control_shader,
-  shaderc_glsl_tess_evaluation_shader
+  // Forced shader kinds. These shader kinds force the compiler to compile the
+  // source code as the specified kind of shader.
+  shaderc_glsl_forced_vertex_shader,
```

Comment:
Yes, what Andrew said. That's more what I expect and want.

Separately:  I like *_infer_from_source better than *_deduce_from_pragma 
For example, in future we might want to support a different source language (non GLSL) that has a more graceful way of indicating shader type in the shader source text.